src/nextjournal/clerk/render.cljs

```diff
@@ -745,25 +744,48 @@
 (defn handle-initial-load [^js _e]
   (history-push-state {:path (subs js/location.pathname 1) :replace? true}))
 
-(defn click->xhr-request [e]
-  (when-some [url (some-> e .-target closest-anchor-parent .-href ->URL)]
+(defn fetch+set-state [edn-path]
+  (js/console.log "fetch" edn-path)
+  (.. ^js (js/fetch edn-path)
+      (then (fn [r]
+               (if (.-ok r)
+                 (.text r)
+                 (throw (ex-info "Not Found" {:response r})))))
+      (then (fn [edn] (set-state! {:doc (read-string edn)}) {:ok true}))
+      (catch (fn [e] (js/console.error "Fetch failed" e) {:ok false :error e}))))
+
+(defn click->fetch [e]
+  (when-some [url (some-> ^js e .-target closest-anchor-parent .-href ->URL)]
     (when-not (ignore-anchor-click? e url)
       (.preventDefault e)
       (let [path (.-pathname url)
             edn-path (str path (when (str/ends-with? path "/") "index") ".edn")]
-        (js/console.log "fetch EDN" edn-path )
-        (-> (js/fetch edn-path)
-            (.then (fn [r]
-                     (if (.-ok r)
-                       (.text r)
-                       (throw (ex-info "Not Found" {:response r})))))
-            (.then (fn [edn]
-                     (set-state! {:doc (read-string edn)})
-                     (.pushState js/history #js {} ""
-                                 (cond-> path
-                                   (not (str/ends-with? path "/"))
-                                   (str "/"))))) ;; trailing slash is needed to make relative paths work
-            (.catch (fn [e] (js/console.error "Fetch failed" e ))))))))
+        (.. (fetch+set-state edn-path)
+            (then (fn [{:keys [ok]}]
```

Comment:
What's the `ok` here? Is it https://developer.mozilla.org/en-US/docs/Web/API/Response/ok? If yes I don't understand why we're accessing it through `:keys`. Should maybe be https://developer.mozilla.org/en-US/docs/Web/API/Response/text instead?

Reply:
I'm already using `.text` and `.-ok` from the fetch response object, but since I catch here, I need to signal failures or success to the consuming `.then` chains. See https://github.com/nextjournal/clerk/blob/69ed81bc8faa68fb49019973e3c09b71b30c09d2/src/nextjournal/clerk/render.cljs#L747-L755